### DISCUSSION TITLE
Error parameter in express middleware

### DISCUSSION THREAD

**Discussion by brandonk3nt** [RobinTail/express-zod-api#1907] [Q&A] [answered]
In express, you can define middleware to handle errors with a fourth parameter `err`:
```
app.use((err, req, res, next) => {
  // Handle err object here
})
```

You can also define this middleware after all the routes as the last item in your express stack. With EZ API you can define express-style middleware, but it does not allow for the fourth `err` parameter in the middleware function. Is this possible with EZ API? What is the best way to implement a "catch-all" error handler, and can I define the order of my middleware like express so that I can run my error middleware last after all my routes?

**Comment by RobinTail** [comment, marked as answer]
In `express-zod-api` that job is carried out by ResultHandler.
The errors happened in endpoint handlers, middlewares, IO validation — will be handled by the ResultHandler assigned to the EndpointsFactory those endpoints produced at.
Parsing errors and `404` errors are handled by the ResultHandler assigned to `errorHandler` in your config (by default it's `defaultResultHandler`), @brandonk3nt

**Comment by RobinTail** [comment]
Related discussion
https://github.com/RobinTail/express-zod-api/discussions/1876
